### DISCUSSION TITLE
HyperSDK connected to IPFS

### DISCUSSION THREAD

**Discussion by Islandersfan2016** [ava-labs/hypersdk#187] [Chat]
After taking a look at the HyperSDK docs, I think it would be possible to build a subnet that's connect to a storage layer. For example, using the HyperSDK for a subnet that allows for metadata to be on-chain. This plan may already be down the road with SDK, but there are other chains just created that have this functionality. HyperSDK though allows for companies to create their own chain with such a functionality. I can see this being a "game-changer" with a number of industries (supply chain, NFTs).

**Comment by patrick-ogrady** [comment]
Adding integrated "blob storage support" is definitely on the roadmap. I played around a bit with this idea when working on the [indexvm](https://github.com/ava-labs/indexvm) but it still relied on the client to interface with IPFS (the chain didn't do anything to improve data redundancy/serve data/check to make sure other validators had data).

Based on that experience, my guess is that the mechanism added to the HyperSDK will, however, be distinct from IPFS. We can use the provided AvalancheGo networking + validator identities to more seamless integrate this functionality (without having to care about objects/connect to peers on the broader IPFS network). There are likely a number of optimizations we can do if we use the chain as a DoS-resistance mechanism for uploading files that IPFS doesn't have the luxury of doing because it is all best-effort.
